octoprint_mrbeam/static/js/settings/camera_settings.js

```diff
@@ -318,6 +320,23 @@ $(function () {
                 }
             );
         };
+
+        self.cameraSettingsActiveChanged = function(newvalue){
+            if (newvalue){
+                self.forceTakeNewPicture()
+                self.imageReloadTimer = setInterval(function () {
+                    self.forceTakeNewPicture()
+                }, 3000);
+            }else{
+                if (self.imageReloadTimer){clearInterval(self.imageReloadTimer)}
+            }
+        }
+
+        self.forceTakeNewPicture = function(){
+            if (self.readyToLaser.lid_fully_open() && !self.readyToLaser.state.interlocksClosed()){
+                OctoPrint.simpleApiCommand("mrbeam", "take_undistorted_picture", {})
```

Comment:
Same here could we make a class or file with all this potential events? it would also help us visualise all the things we use from octoprint. We could start by creating the class and adding this one event on this PR and slowly add more or create a ticket to research other potential events